src/index.ts

```diff
@@ -87,6 +87,29 @@ const run = async () => {
 	await summary.write();
 };
 
+function handleError(error: any, kind: string, permission: string) {
+	if (
+		error instanceof RequestError
+	) {
+		switch (error.status) {
+			case 403:
+			case 404:
+				core.warning(
+					`Couldn't write a comment to the ${kind}. Please make sure your job has the permission '${permission}: write'.\n` +
```

Comment:
As noted the use of `: write` fixes the case where the code incorrectly suggested a need for `: read`...